### PR TITLE
Add fold tests for Cypher in test_compiler

### PR DESCRIPTION
**Pull request by LWprogramming**
Right now in PR #437 I'm working on getting the fold functionality to
work. This PR contains the majority of compiler tests for fold for
Cypher, and can be reviewed as soon as PRs #459, #463, and #464 are
merged into master.

Four fold-related tests are still skipped for now: these tests include
optional in the same GraphQL query as fold and there may be some
potential edge cases I haven't fully fleshed out yet. Those tests are
already written; I've just left them out so that we can merge these
first and merge those tests along with the correct implementation for
OPTIONAL once that's taken care of.

Also labeled all the fold-related `SKIP_TESTS` so we know why they're skipped.

### CODE REVIEW

File: graphql_compiler/tests/test_compiler.py [hunk 1]
```diff
@@ -4043,7 +4105,28 @@ def test_traverse_and_fold_and_traverse(self):
...
+              (Animal__in_Animal_ParentOf__out_Animal_ParentOf___1)-[:Animal_OfSpecies]->
+              (Animal__in_Animal_ParentOf__out_Animal_ParentOf__out_Animal_OfSpecies___1:Species)
+            WITH
+              Animal___1 AS Animal___1,
+              Animal__in_Animal_ParentOf___1 AS Animal__in_Animal_ParentOf___1,
+              collect(Animal__in_Animal_ParentOf__out_Animal_ParentOf___1) AS
+                collected_Animal__in_Animal_ParentOf__out_Animal_ParentOf___1,
+              collect(Animal__in_Animal_ParentOf__out_Animal_ParentOf__out_Animal_OfSpecies___1) AS
+                collected_Animal__in_Animal_ParentOf__out_Animal_ParentOf__out_Animal_OfSpecies___1
+          RETURN
```
Comment (obi1kenobi) [inline graphql_compiler/tests/test_compiler.py]: The indentation appears off here, `RETURN` and `MATCH` are not at the same indentation level.

File: graphql_compiler/tests/test_compiler.py [hunk 2]
```diff
@@ -4486,7 +4662,22 @@ def test_filter_within_fold_scope(self):
             ])}
         '''
         expected_sql = NotImplementedError
-        expected_cypher = SKIP_TEST
+        expected_cypher = '''
+            MATCH (Animal___1:Animal)
+            OPTIONAL MATCH (Animal___1)-[:Animal_ParentOf]->(Animal__out_Animal_ParentOf___1:Animal)
+            WHERE (
```
Comment (obi1kenobi) [inline graphql_compiler/tests/test_compiler.py]: Isn't the `WHERE` indented in the other examples? Let's try to be consistent.
